torchci/scripts/check_alerts.py

```diff
@@ -444,8 +444,16 @@ def handle_flaky_tests_alert(existing_alerts: List[Dict]) -> Dict:
     return None
 
 
-def check_for_recurrently_failing_jobs_alert(repo: str, branch: str, dry_run: bool):
+# filter job names that don't match the regex
+def filter_job_names(job_names: List[str], job_name_regex: str) -> List[str]:
+    if job_name_regex:
+        return list(filter(lambda job: re.match(job_name_regex, job), job_names))
```

Comment:
Please use list comprehensions, which are imo easier to read but also faster:
```python
>>>from timeit import timeit
>>> timeit("list(filter(lambda x:x%2 == 1, range(100)))")
4.687839625000009
>>> timeit("[x for x in range(100) if x%2 == 1]")
3.685032250000006
```
```suggestion
        return [job_name for job_name in job_names if re.match(job_name_regex, job_name)]
```